src/Microsoft.Diagnostics.Monitoring.EventPipe/DiagnosticsMonitor.cs

```diff
@@ -12,7 +12,7 @@ namespace Microsoft.Diagnostics.Monitoring.EventPipe
 {
     partial class DiagnosticsEventPipeProcessor
     {
-        private sealed class DiagnosticsMonitor : IAsyncDisposable
+        internal sealed class DiagnosticsMonitor : IAsyncDisposable
```

Comment:
Since this is now a shared piece of logic, we should rename it and no longer make it a nested class.

Reply:
Name suggestion: EventPipeSessionReader

Reply:
I renamed it to EventPipeSessionManager to better reflect that it is creating and managing the state of the EventPipeSession.